src/main/java/io/github/sefiraat/networks/slimefun/network/NetworkQuantumStorage.java

```diff
@@ -279,18 +326,26 @@ private QuantumCache addCache(@Nonnull BlockMenu blockMenu) {
         final String voidString = BlockStorage.getLocationInfo(location, BS_VOID);
         final int amount = amountString == null ? 0 : Integer.parseInt(amountString);
         final boolean voidExcess = voidString == null || Boolean.parseBoolean(voidString);
+        // We use either the maximum amount if custom is not set, or the custom amount if it is set and is supported for this quantum
+        int customMaxAmount = this.maxAmount;
+        if (this.supportsCustomMaxAmount) {
+            final String customMaxAmountString = BlockStorage.getLocationInfo(blockMenu.getLocation(), BS_CUSTOM_MAX_AMOUNT);
+            if (customMaxAmountString != null) {
+                customMaxAmount = Integer.parseInt(customMaxAmountString);
+            }
+        }
         final ItemStack itemStack = blockMenu.getItemInSlot(ITEM_SLOT);
 
-        QuantumCache cache = createCache(itemStack, blockMenu, amount, voidExcess);
+        QuantumCache cache = createCache(itemStack, blockMenu, amount, voidExcess, customMaxAmount);
 
         CACHES.put(location, cache);
         return cache;
     }
 
-    private QuantumCache createCache(@Nullable ItemStack itemStack, @Nonnull BlockMenu menu, int amount, boolean voidExcess) {
+    private QuantumCache createCache(@Nullable ItemStack itemStack, @Nonnull BlockMenu menu, int amount, boolean voidExcess, int customMaxAmount) {
```

Comment:
Shouldn't this just be called max amount?
I'd also order this directly after amount